File: bytedance/415.py
# 415. 字符串相加

class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        res = ""
        i, j, carry = len(num1) - 1, len(num2) - 1, 0
        while i >= 0 or j >= 0: # 字符串长短操作
            n1 = int(num1[i]) if i >= 0 else 0
            n2 = int(num2[i]) if j >= 0 else 0
            tmp = n1 + n2 + carry
            carry = tmp // 10 # 进位操作
            res = str(tmp % 10) + res
            i, j = i - 1, j - 1
        return "1" + res if carry else res

x = Solution()
# print(x.addStrings("199", "3211"))
print(x.addStrings("1", "9"))
print(x.addStrings("1", "7"))
print(x.addStrings("4", "9"))

        

File: bytedance/dy/415.java
// 415. 字符串相加

// 长度判断
class Solution {
    public String addStrings(String num1, String num2) {
        int acc = 0;
        int i = num1.length();
        int j = num2.length();

        StringBuilder ans = new StringBuilder();
        while (i >= 0 || j >= 0 || acc > 0) {
            int x = i >= 0 ? num1.charAt(i) - '0' : 0;
            int y = j >= 0 ? num2.charAt(j) - '0' : 0;
            int res = x + y + acc;
            ans.append(res % 10);
            acc = res / 10;
            i--;
            j--;
        }
        ans.reverse();
        return ans.toString();
    }
}